orchagent/tunneldecaporch.cpp

```diff
@@ -1019,3 +1462,104 @@ bool TunnelDecapOrch::getQosMapId(const std::string &tunnelKey, const std::strin
     }
     return true;
 }
+
+void TunnelDecapOrch::updateUnhandledDecapTunnelTerms(const string &tunnel_name, const string &src_ip_str)
+{
+    SWSS_LOG_ENTER();
+    SWSS_LOG_INFO("Updating unhandled decap tunnel terms for tunnel %s with source IP %s",
+                  tunnel_name.c_str(), src_ip_str.c_str());
+
+    auto tunnel_it = unhandledDecapTerms.find(tunnel_name);
+    if (tunnel_it != unhandledDecapTerms.end())
+    {
+        for (auto term_it = tunnel_it->second.begin(); term_it != tunnel_it->second.end(); ++term_it)
+        {
+            auto &term = term_it->second;
+            if (term.term_type == TUNNEL_TERM_TYPE_P2P || term.term_type == TUNNEL_TERM_TYPE_MP2MP)
```

Comment:
Suggest checking `!src_ip_str.empty()`

Reply:
As the above, we can discuss this offline.